src/components/OrdersWidget/index.tsx

```diff
@@ -56,11 +56,18 @@ function classifyOrders(
   orders: AuctionElement[],
   state: FilteredOrdersState,
   ordersType: 'orders' | 'pendingOrders',
-): void {
+  alreadyCounted?: Set<string>,
+): Set<string> {
   const now = new Date()
   const isOrderActiveFn = ordersType === 'pendingOrders' ? isPendingOrderActive : isOrderActive
+  const counted = new Set<string>()
 
   orders.forEach(order => {
+    // don't process already processed orders
+    if (alreadyCounted && alreadyCounted.has(order.id)) return
```

Comment:
I didn't understand why an order would be two times.

It's pending, but once we get the confirmation the changes into a not pending one.

Reply:
If we do this only in the widget, wouldn't we have this problem in any part using the global state?

Would it make sense to just delete the pending orders once we have the confirmation?

Reply:
We delete it on confirmation of mined tx already. I don't understand what you mean.

@alfetopito described exactly why we have duplicate orders in the issue